httpbin/handlers.go

```diff
@@ -464,7 +464,7 @@ func (h *HTTPBin) DeleteCookies(w http.ResponseWriter, r *http.Request) {
 			Expires:  time.Now().Add(-1 * 24 * 365 * time.Hour),
 		})
 	}
-	w.Header().Set("Location", "/cookies")
+	w.Header().Set("Location", h.prefix+"/cookies")
 	w.WriteHeader(http.StatusFound)
```

Comment:
Given how often this pattern now shows up across handlers, I propose that we wrap the logic up in a helper method:

```golang
func (h *HTTPBin) doRedirect(path string, code int) {
    w.Header().Set("Location", h.prefix+"/cookies")
    w.WriteHeader(code)
}
```

(That name would clash with the _existing_ `doRedirect()` helper, which is a bit higher level and serves only the `Redirect`/`AbsoluteRedirect`/`RelativeRedirect` handler.  I'd suggest we rename the existing `doRedirect` to something like `handleRedirect` instead.)

Reply:
You mean a `doRedirectCookies` method or `cookies` a parameter to some method?

Reply:
I mean there are now a bunch of places where we've got code like this, where we're constructing a redirect URL and must remember to include the prefix, so I'd like to have a helper method like the one I showed above.

_This_ code would then become something like

```go
h.doRedirect("/cookies", http.StatusFound)
return
```

and the new `doRedirect()` helper would take care of handling the prefix.

Where it gets messy is that there's already an **existing**, higher level helper method `doRedirect()` helper that will need to be renamed.  I think renaming the existing helper to `handleRedirect()` makes sense, as that better captures what the existing helper is actually doing.

Reply:
I renamed the `doRedirect` to `handleRedirect` and used at two places. Is there a reason why https://github.com/waschik/go-httpbin/blob/7e81a9365c6fe9f1a4691742792dcd2e3d8a1fa6/httpbin/handlers.go#L883 is using `r.URL.String()` and not `r.URL.Path`? If `r.URL.Path` would be okay it would be possible to replace it also with the new `doRedirect`. For `r.URL.String()` I am not sure if it will not sometimes include the schema and then prefix might be wrong.

For the absolute redirects new `doRedirect` would also not possible otherwise I get locations like `/a-prefixhttp://...`

Reply:
> Is there a reason why https://github.com/waschik/go-httpbin/blob/7e81a9365c6fe9f1a4691742792dcd2e3d8a1fa6/httpbin/handlers.go#L883 is using `r.URL.String()` and not `r.URL.Path`?

I _think_ this is to ensure that any query params are carried through the redirects, but it looks like we don't have any explicit test cases covering that behavior.  Let's keep it as-is for now.

> For the absolute redirects new `doRedirect` would also not possible otherwise I get locations like `/a-prefixhttp://...`

Ah that's a good point.  Maybe he new `doRedirect()` should change its behavior based on whether the given argument starts with `http://` or `https://`?

Reply:
Maybe it works if prefix is only added if arguments start with `/`. I'll experiment a bit with this. Otherwise option would be to parse URL or add another parameter.

Reply:
Yep, I think pivoting on whether the first character is `/` would be fine here!

Reply:
I implemented this with slash checking. Wrong adding of prefix of external prefix was shown in test cases. Hope it is correct now.